web/src/message_list_view.js

```diff
@@ -210,6 +210,12 @@ function populate_group_from_message_container(group, message_container) {
         group.topic_is_resolved = resolved_topic.is_resolved(group.topic);
         group.topic_muted = user_topics.is_topic_muted(group.stream_id, group.topic);
         group.topic_unmuted = user_topics.is_topic_unmuted(group.stream_id, group.topic);
+        group.development = page_params.development_environment;
+        group.visibility_policy = user_topics.get_topic_visibility_policy(
+            group.stream_id,
+            group.topic,
+        );
+        group.all_visibility_policies = user_topics.all_visibility_policies;
```

Comment:
I don't like adding `development` and `all_visibility_policies` to this object -- those are sorta global properties, and having a copy of them for each recipient group just feels weird.

Maybe it's hard to fix, and it's not exactly important, but it would be nice if we could have these properties just passed into template rendering.

Reply:
(But I guess `all_visibility_policies` shouldn't be a thing that needs to in the DOM at all?)

Reply:
I think an acceptable option would be to add a blank line followed by a comment above those two fields noting they are not specific to this group but that this is the easiest way we've figured out for passing the data to the template rendering.

Reply:
Added comment